src/components/Skeleton/Skeleton.tsx

```diff
@@ -28,27 +28,31 @@ interface SkeletonProps extends VibeComponentProps {
 }
 
 const Skeleton: FC<SkeletonProps> & {
-  types?: typeof SKELETON_TYPES;
+  types?: typeof SkeletonType;
   sizes?: typeof SKELETON_SIZES;
 } = ({
-  type = SKELETON_TYPES.RECTANGLE,
+  type = SkeletonType.RECTANGLE,
   size = SKELETON_SIZES.CUSTOM,
   className,
   wrapperClassName,
   width,
   height,
   fullWidth = false
 }) => {
-  const skeletonType = SKELETON_ALLOWED_TYPES.indexOf(type) > -1 ? type : SKELETON_TYPES.RECTANGLE;
-  const typeDescription = skeletonType.toUpperCase() as keyof typeof SKELETON_ALLOWED_SIZES;
-  const skeletonSize = SKELETON_ALLOWED_SIZES[typeDescription].indexOf(size) > -1 ? size : SKELETON_CUSTOM_SIZE;
+  const skeletonType = Object.values(SkeletonType).includes(type) ? type : SkeletonType.RECTANGLE;
+
+  // Skeleton has sizes only for text type, other types support only custom size
+  const skeletonSize = (Object.values(TextSkeletonSize) as string[]).includes(size) ? size : SKELETON_CUSTOM_SIZE;
   return (
     <div className={cx(SKELETON_CSS_BASE_CLASS, wrapperClassName, { [styles.fullWidth]: fullWidth })}>
       <div
         className={cx(
           bemHelper({ element: skeletonType }),
           bemHelper({ element: skeletonType, state: skeletonSize }),
-          className
+          className,
+          {
+            [styles.fullWidth]: bemHelper({ element: skeletonType, state: "full-width" })
```

Comment:
got confused here... why bemHelper result is a condition for applying `styles.fullWidth`?

Reply:
you righttt i will fix